
```{r}
library(qiime2R)
library(tidyr)
library(microbiome)
library(compositions)
library(reshape2)
library(gplots)
library(RColorBrewer)
library(ggplot2)
library(janitor)
library(randomcoloR)
library(dplyr)
library(roperators)
library(ggpubr)
library(wesanderson)

```

```{r}

#This code block is to stage all the files needed for visualization


#Stage taxonomy file using Qiime2
taxa_vsearch <- parse_taxonomy(taxonomy = read_qza('../taxonomy_vsearch/classification.qza')$data)
# taxa_blast <- parse_taxonomy(taxonomy = read_qza('../taxonomy_blast/classification-blast.qza')$data)

#Stage metadata file
sample_metadata <- read.csv('../read-files/metadata.tsv', sep='\t')

#Stage unstratified/strat kegg pathway files
unstrat_kegg <- read.csv('../picrust2-strat/out/KEGG_pathways_out/path_abun_unstrat.tsv.gz', sep = '\t')
strat_kegg <- read.csv('../picrust2-strat/out/KEGG_pathways_out/path_abun_contrib.tsv.gz', sep = '\t')

#Stage unstrat/strat KEGG ortholog files 
ko_unstrat <- read.csv('../picrust2-strat/out/KO_metagenome_out/pred_metagenome_unstrat.tsv.gz', sep = '\t')
ko_strat <- read.csv('../picrust2-strat/out/KO_metagenome_out/pred_metagenome_contrib.tsv.gz', sep = '\t')

#Open selected KOs (for plotting relevant Kegg pathways)
deg_kos <- read.csv('./select_kos.txt', sep = '\t', header = TRUE)
rownames(deg_kos) <- deg_kos[,1]
deg_kos[,1] <- NULL

#open the pathway to ko file
pth_2_ko <- read.csv('~/bioinfo_pipelines/picrust2-2.4.2/picrust2/default_files/pathway_mapfiles/KEGG_pathways_to_KO.tsv', header =FALSE, sep='\t') 
#Convert empty strings as NA
pth_2_ko[pth_2_ko==""] <- NA 

#Filter to only Select KOs

pth_2_ko <- pth_2_ko[pth_2_ko$V1 %in% rownames(deg_kos),]
rownames(pth_2_ko) <- pth_2_ko[,1]
pth_2_ko[,1] <- NULL

#Retrieve Sample ids so I can call it as a variable 
sample_ids <- c('HC.DS','HC.XY','HC.HX','HC.HD')

cons_codes <- c('HC-DS'='B2',
                'HC-XY'='B7',
                'HC-HX'='B1',
                'HC-HD'='A5')

strat_kegg <- strat_kegg %>% mutate(sample=recode(sample, !!!cons_codes))


```

```{r}


samps <- c('HC-DS','HC-XY','HC-HX','HC-HD')


#This code block is to load the required files to plot NSTI vs relative abundance
abs_count_otu <-  read_qza(file = '../dada2/table.qza')$data
rel_count_otu <- as.data.frame(apply(abs_count_otu,2,function(x){x/sum(x)}))
nsti_asvs <- as.data.frame(read.csv('../picrust2-strat/out/marker_predicted_and_nsti.tsv.gz', sep='\t', row.names = 1))


rel_nsti <- merge(nsti_asvs,rel_count_otu,by="row.names",all.x=TRUE)

rel_nsti <- melt(rel_nsti, measure.vars = samps, variable.name = 'sample.id', value.name = 'rel.abundance')


#Group accding to consortia



ggplot_rel_plot <- ggplot(data=rel_nsti,mapping= aes(x=metadata_NSTI, y=rel.abundance, color=sample.id)) + geom_point()

plot(ggplot_rel_plot)

#No need to present this. This is just a diagnostic plot
```


```{r}
#This codeblock is for concatenation and export of feature counts together with their taxonomies


Feature_counts_taxa <- cbind(taxa_vsearch, rel_count_otu * 100) 

fam_counts_sum <- Feature_counts_taxa %>% group_by(Family) %>% summarize_at(samps, sum)

genus_counts_sum <- Feature_counts_taxa %>% group_by(Genus) %>% summarize_at(samps, sum)

class_counts_sum <- Feature_counts_taxa %>% group_by(Class) %>% summarize_at(samps, sum)

write.table(Feature_counts_taxa, '../dada2/Summary_taxa_feature_table.tsv', sep ='\t', col.names = T, row.names = T)
write.table(genus_counts_sum, '../dada2/Summary_genus_feature_table.tsv', sep ='\t', col.names = T, row.names = T)

```

```{r}
#Generate table for unstratified relative abundances
dir.create('./intermediate_files')

row.names(unstrat_kegg) <- unstrat_kegg$pathway
unstrat_kegg_no_desc <- unstrat_kegg[,2:ncol(unstrat_kegg)]


unstr_keg_mat <- data.matrix(unstrat_kegg_no_desc,rownames.force = TRUE)
#get relative abundance using adorn_percentages  (From Janitor package)
rel_abun_unstrkeg <- adorn_percentages(unstrat_kegg_no_desc, denominator = "col")

write.table(rel_abun_unstrkeg, file = './intermediate_files/unstratified_relative_abundance_KEGG.tsv', sep ='\t', col.names = T)

```


```{r}
dir.create('./out')

gen_mat <- function(df, ids){
  df_mat <- as.matrix(select(df, ids))
  return(df_mat)
}

#Generate Heatplot from unstratified KEGG plots, filtered only for xenobiotic degradation

heat_palette <- colorRampPalette(brewer.pal(3,'YlOrBr'))(n = 100)
#Filter unstrat based on deg_kos table

unstr_keg_filtered <- subset(unstrat_kegg_no_desc, unstrat_kegg$pathway %in% row.names(deg_kos)) 




#Plot heatmap based only on degradation
pdf(file = "./KEGG-pathway-heatmap.pdf",   
    width = 8, 
    height = 11) 

heatmap.2(gen_mat(unstr_keg_filtered,sample_ids), 
          labRow = unstr_keg_filtered$description, 
          key=TRUE, keysize=1, symkey=T, lhei = c(1,7), 
          density.info = "none", dendrogram ="none", trace='none', scale='row',
          
          col = heat_palette, margins=c(10,24))

dev.off()
```

```{r}
#Generate Heatplot from unstratified KOs per pathway in deg_kos
#I'll generate one first then convert that to a function


#This works na, what I did was generate a list of dataframes first then concatenated them afterwards. I can reuse the degdfs to generate individual 



degdf_list <- list()

for(i in 1:nrow(pth_2_ko)){
  fun_list <- na.omit(unlist(pth_2_ko[i,]))
  fun_df <- subset.data.frame(ko_unstrat, function. %in% fun_list)
  fun_df$pathway <- rownames(pth_2_ko[i,])
  fun_df$pathway_desc <- deg_kos[fun_df$pathway,]
  degdf_list[[i]] <-fun_df
}

#Concatenate all dfs in degdf list
deg_df = do.call(rbind, degdf_list)
#The pathway column needs to be categorical pala para mag-work lol

#Sort the pathways also in alphabetical order
sorted_pathways <- sort(unique(deg_df$pathway_desc))
deg_df$pathway <- factor(deg_df$pathway)
deg_df$pathway_desc <- factor(deg_df$pathway_desc, levels=sorted_pathways)
#reset index
rownames(deg_df) <- NULL

#Let's remove yung atrazine, dioxin, and styrene pathways
deg_df  <- subset(deg_df, !(pathway %in% c('ko00791','ko00621' ,'ko00643')))

```

```{r}

#Define my color palettes
#I just copied this from some guy's Stackx answer haha
c16 <- c("dodgerblue2", "#E31A1C", "green4", "#6A3D9A", "#FF7F00", "black", "gold1", "skyblue2", "palegreen2", "#FDBF6F", "gray70", "maroon", "orchid1", "darkturquoise", "darkorange4", "brown") 




#Set Pathways as factor

pdf(file = "./out/Per-gene-heatmap.pdf",   
    width = 10, 
    height = 20) 

heatmap.2((gen_mat(deg_df, sample_ids)), 
          labRow = deg_df$description, labCol = sample_metadata$treatment, 
           key=TRUE, keysize=1, symkey=T, lhei = c(1,17), 
          Rowv = FALSE, Colv = FALSE,
          density.info = "none", dendrogram ="none", trace='none', scale='row',
          margins=c(10,50),
          col = heat_palette,
          cexRow = 0.7,
          RowSideColors=c16[as.numeric(deg_df$pathway_desc)], na.color = 'black' )

legend("left",      
    legend = unique(deg_df$pathway_desc),
    col = c16[unique(deg_df$pathway_desc)], 
    inset = c(-0.09,0),
    lty= 1,             
    lwd = 5,
    pch =14,
    cex=.45,
    xpd=T
    )

dev.off()

```




```{r}
#Generate table for unstratified relative abundances
dir.create('./intermediate_files')

row.names(unstrat_kegg) <- unstrat_kegg$pathway
unstrat_kegg_no_desc <- unstrat_kegg[,2:ncol(unstrat_kegg)]


unstr_keg_mat <- data.matrix(unstrat_kegg_no_desc,rownames.force = TRUE)
#get relative abundance using adorn_percentages  (From Janitor package)
rel_abun_unstrkeg <- adorn_percentages(unstrat_kegg_no_desc, denominator = "col")

write.table(rel_abun_unstrkeg, file = './intermediate_files/unstratified_relative_abundance_KEGG.tsv', sep ='\t', col.names = T)

```


```{r}
dir.create('./out')

gen_mat <- function(df, ids){
  df_mat <- as.matrix(select(df, ids))
  return(df_mat)
}

#Generate Heatplot from unstratified KEGG plots, filtered only for xenobiotic degradation

heat_palette <- colorRampPalette(brewer.pal(3,'YlOrBr'))(n = 100)
#Filter unstrat based on deg_kos table

unstr_keg_filtered <- subset(unstrat_kegg_no_desc, unstrat_kegg$pathway %in% row.names(deg_kos)) 




#Plot heatmap based only on degradation
pdf(file = "./out/pathway-heatmaps/KEGG-pathway-heatmap.pdf",   
    width = 8, 
    height = 11) 

heatmap.2(gen_mat(unstr_keg_filtered,sample_ids), 
          labRow = unstr_keg_filtered$description, 
          key=TRUE, keysize=1, symkey=T, lhei = c(1,7), 
          density.info = "none", dendrogram ="none", trace='none', scale='row',
          
          col = heat_palette, margins=c(10,24))


dev.off()
```



```{r}

#Define my color palettes
#I just copied this from some guy's Stackx answer haha
c16 <- c("dodgerblue2", "#E31A1C", "green4", "#6A3D9A", "#FF7F00", "black", "gold1", "skyblue2", "palegreen2", "#FDBF6F", "gray70", "maroon", "orchid1", "darkturquoise", "darkorange4", "brown") 




#Set Pathways as factor

pdf(file = "./out/pathway-heatmaps/Per-gene-heatmap.pdf",   
    width = 10, 
    height = 20) 

heatmap.2((gen_mat(deg_df, sample_ids)), 
          labRow = deg_df$description, labCol = sample_metadata$treatment, 
           key=TRUE, keysize=1, symkey=T, lhei = c(1,17), 
          Rowv = FALSE, Colv = FALSE,
          density.info = "none", dendrogram ="none", trace='none', scale='row',
          margins=c(10,50),
          col = heat_palette,
          cexRow = 0.7,
          RowSideColors=c16[as.numeric(deg_df$pathway_desc)], na.color = 'black' )

legend("left",      
    legend = unique(deg_df$pathway_desc),
    col = c16[unique(deg_df$pathway_desc)], 
    inset = c(-0.09,0),
    lty= 1,             
    lwd = 5,
    pch =14,
    cex=.45,
    xpd=T
    )

dev.off()

```

```{r}

#Generate Cumulative relative abundances per consortia for specific degradation pathways

#Generate sorting vector for the sample ids
sort_cols = c('A5', 'B1', 'B2', 'B7')
cons_codes <- c('HC.DS'='B2',
                'HC.XY'='B7',
                'HC.HX'='B1',
                'HC.HD'='A5')

#Filter then convert to long form so it's easier to plot
filt_rel_abun <- subset(rel_abun_unstrkeg, rownames(rel_abun_unstrkeg) %in% rownames(deg_kos)) %>% melt(measure.vars = sample_ids, variable.name = 'sample.id', value.name ='rel.abundance') %>% mutate(across(description, factor))  %>% mutate(sample.id = recode(sample.id,!!!cons_codes)) %>% mutate(sample.id=factor(sample.id, levels=sort_cols)) %>%
  mutate(rel.abundance = rel.abundance * 100) #Convert to percent
#Convert to percent

write.table(filt_rel_abun, file =  './intermediate_files/out-Deg-pathway-relabunds.tsv', sep ='\t', col.names = T)



colors <- brewer.pal(12, "Paired")


pdf(file='./out/Deg-pathway-relabunds.pdf', width = 8, height = 5)


plot3 <- ggplot(filt_rel_abun, aes(x=sample.id, y=rel.abundance, fill=description)) +   geom_bar(position="stack", stat="identity") +
labs(title='Degradation pathway relative abundances', x='Consortia code',y='% relative pathway abundances') +scale_fill_manual(values=colors) + labs(color='Degradation pathway description')


plot(plot3)
dev.off()


plot(plot3)

ggsave(file='./out/pathway-counts/Deg_pathway_relabunds.svg',plot3, device = 'svg', width=8, height=5)

```

```{r}

#This code block is to summarize how many Kegg orthologs are identified for this study. This is based on the database used by PICRUST2  which may have been curated to remove double entries.
ko_counts <- group_by(deg_df, pathway_desc)
ko_counts %>% count(pathway_desc)
```




```{r}
#This code block is to visualize the generated filtered table above 


fam_list <- c(unique(kegg_strat_relcounts$taxon))
fam_cols  <- distinctColorPalette(20)

pdf(file = "./out/Per-Family-KEGG-Relcounts.pdf",   
    width = 12, 
    height = 5) 


#This function plots all the stratified abundances in a single row
plot4<- ggplot(kegg_strat_relcounts, aes(x=description, y=taxon_rel_function_abun, fill = taxon)) +
geom_bar(position="stack", stat="identity")  +   
scale_fill_manual(values=fam_cols, drop=T) +
labs(y= "Function relative abundance (Per taxon)", x="HC Degradation Pathways (Level 3 KEGG)")+ coord_flip() + facet_wrap(~sample, nrow = 1)


plot(plot4)
dev.off()



plot(plot4)

```

```{r}

kegg_strat_relcounts_order <- strat_kegg %>% transmute(description = as.factor(description), sample, taxon = taxa_vsearch[taxon, 'Order'], taxon_rel_function_abun) %>% group_by(sample, description,taxon) %>% summarise(across(taxon_rel_function_abun, sum)) %>% filter(description %in% deg_df$pathway_desc) #Summarise based on taxon_rel 

fam_cols  <- distinctColorPalette(20)

pdf(file = "./out/Per-order-relcounts-KEGG.pdf",   
    width = 14, 
    height = 7) 

plot5<- ggplot(kegg_strat_relcounts_order, aes(x=description, y=taxon_rel_function_abun, fill = taxon)) +
geom_bar(position="stack", stat="identity")  +   
scale_fill_manual(values=fam_cols, drop=T) +
labs(y= "Function relative abundance (Per taxon)", x="HC Degradation Pathways (Level 3 KEGG)")+ coord_flip() + facet_wrap(~sample, nrow = 1)

plot(plot5)
dev.off()

plot(plot5)

```



```{r}

kegg_strat_relcounts_class <- strat_kegg %>% transmute(description = as.factor(description), sample, taxon = taxa_vsearch[taxon, 'Class'], taxon_rel_function_abun) %>% group_by(sample, description,taxon) %>% summarise(across(taxon_rel_function_abun, sum)) %>% filter(description %in% deg_df$pathway_desc) #Summarise based on taxon_rel 


pdf(file = "./out/Per-class-relcounts-KEGG.pdf",   
    width = 14, 
    height = 7) 

plot7 <- ggplot(kegg_strat_relcounts_class, aes(x=description, y=taxon_rel_function_abun, fill = taxon)) +
geom_bar(position="stack", stat="identity")  +   
scale_fill_manual(values=fam_cols, drop=T) +
labs(y= "Function relative abundance (Per taxon)", x="HC Degradation Pathways (Level 3 KEGG)")+ coord_flip() + facet_wrap(~sample, nrow = 1)


plot(plot7)
ggsave(file='./out/pathway-counts/Per-class-relcounts-KEGG.svg',plot7, device = 'svg', width=10, height=4)

```



```{r}

fam_cols  <- distinctColorPalette(20)


kegg_strat_relcounts_unfiltered_fam <- strat_kegg %>% transmute(description = as.factor(description), sample, taxon = taxa_vsearch[taxon, 'Family'], taxon_rel_function_abun) %>% group_by(sample, description,taxon) %>% summarise(across(taxon_rel_function_abun, sum))  #Summarise based on taxon_rel 


pdf(file = "./out/pathway-counts/unfiltered-relcounts-KEGG-Family.pdf",   
    width = 10, 
    height = 20) 

plot8 <- ggplot(kegg_strat_relcounts_unfiltered_fam, aes(x=description, y=taxon_rel_function_abun, fill = taxon)) +
geom_bar(position="stack", stat="identity")  +   
scale_fill_manual(values=fam_cols, drop=T) +
labs(y= "Function relative abundance (Per taxon)", x="Pathway Systems (Level 3 KEGG)")+ coord_flip() + facet_wrap(~sample, nrow = 1)

plot(plot8)
dev.off()

plot(plot8)

```



```{r}
#Generate the plot for relative counts
fam_cols  <- distinctColorPalette(20) #The palette below is the palette I opted to use
fam_cols <- c("#BC6FE2", "#7CAFE0", "#7E9991", "#D0E4B5", "#6FE3A5", "#D1E5E5", "#E54CD7", "#DCB192", "#E166B1", "#76DED7", "#DF646F" ,"#CDD576", "#DA9FDA" ,"#6B7DE2", "#913AE2" ,"#836590" ,"#D3E345", "#D7BDD3", "#75E25F", "#E59B46")



kegg_strat_relcounts_phylum <- strat_kegg %>% transmute(description = as.factor(description), sample, taxon = taxa_vsearch[taxon, 'Phylum'], taxon_rel_function_abun) %>% group_by(sample, description,taxon) %>% summarise(across(taxon_rel_function_abun, sum)) %>% filter(description %in% deg_df$pathway_desc)

pdf(file = "./out/Per-phylum-relcounts-KEGG.pdf",   
    width = 14, 
    height = 7) 

plot6<- ggplot(kegg_strat_relcounts_phylum, aes(x=description, y=taxon_rel_function_abun, fill = taxon)) +
geom_bar(position="stack", stat="identity")  +   
scale_fill_manual(values=fam_cols, drop=T) +
labs(y= "Function relative abundance (Per taxon)", x="HC Degradation Pathways (Level 3 KEGG)")+ coord_flip() + facet_wrap(~sample, nrow = 1)
plot6$data$description <- factor(plot6$data$description, levels=sort(unique(plot6$data$description), decreasing=T))


kegg_strat_relcounts_genus <- strat_kegg %>% transmute(description = as.factor(description), sample, taxon = ifelse(is.na(taxa_vsearch[taxon,'Genus'])==TRUE,taxa_vsearch[taxon, 'Family'], taxa_vsearch[taxon,'Genus']), taxon_rel_function_abun) %>% group_by(sample, description,taxon) %>% summarise(across(taxon_rel_function_abun, sum)) %>% filter(description %in% deg_df$pathway_desc)  


pdf(file = "./out/Per-GenusFam-relcounts-KEGG.pdf",   
    width = 14, 
    height = 7) 

plot9 <- ggplot(kegg_strat_relcounts_genus, aes(x=description, y=taxon_rel_function_abun, fill = taxon)) +
geom_bar(position="stack", stat="identity")  +   
scale_fill_manual(values=fam_cols, drop=T) +
labs(y= "Function relative abundance (Per taxon)", x="HC Degradation Pathways (Level 3 KEGG)")+ coord_flip() + facet_wrap(~sample, nrow = 1)

plot9$data$description <- factor(plot9$data$description, levels=sort(unique(plot9$data$description), decreasing=T))



pdf(file = "./out/Stacked-GenusFam-relcounts-KEGG.pdf",   
    width = 11, 
    height = 8) 

gen_fam_plot <- ggarrange(plot6, plot9, nrow = 2, align ='v')

ggsave('./out/pathway-counts/Stacked-GenusFam-relcounts-KEGG.svg', gen_fam_plot,device='svg', width=15, height=10)
```

```{r}


#Stage taxonomy file using Qiime2
taxa_vsearch <- parse_taxonomy(taxonomy = read_qza('../taxonomy_vsearch/classification.qza')$data)
# taxa_blast <- parse_taxonomy(taxonomy = read_qza('../taxonomy_blast/classification-blast.qza')$data)

#Stage metadata file
sample_metadata <- read.csv('../read-files/metadata.tsv', sep='\t')

#Stage unstratified/strat kegg pathway files
unstrat_kegg <- read.csv('../picrust2-strat/out/KEGG_pathways_out/path_abun_unstrat.tsv.gz', sep = '\t')
strat_kegg <- read.csv('../picrust2-strat/out/KEGG_pathways_out/path_abun_contrib.tsv.gz', sep = '\t')

#Stage unstrat/strat KEGG ortholog files 
ko_unstrat <- read.csv('../picrust2-strat/out/KO_metagenome_out/pred_metagenome_unstrat.tsv.gz', sep = '\t')
ko_strat <- read.csv('../picrust2-strat/out/KO_metagenome_out/pred_metagenome_contrib.tsv.gz', sep = '\t')

#Open selected KOs (for plotting relevant Kegg pathways)
deg_kos <- read.csv('./select_kos.txt', sep = '\t', header = TRUE)
rownames(deg_kos) <- deg_kos[,1]
deg_kos[,1] <- NULL

#open the pathway to ko file
pth_2_ko <- read.csv('~/bioinfo_pipelines/picrust2-2.4.2/picrust2/default_files/pathway_mapfiles/KEGG_pathways_to_KO.tsv', header =FALSE, sep='\t') 
#Convert empty strings as NA
pth_2_ko[pth_2_ko==""] <- NA 

#Filter to only Select KOs

pth_2_ko <- pth_2_ko[pth_2_ko$V1 %in% rownames(deg_kos),]
rownames(pth_2_ko) <- pth_2_ko[,1]
pth_2_ko[,1] <- NULL

#Retrieve Sample ids so I can call it as a variable 
sample_ids <- c('HC.DS','HC.XY','HC.HX','HC.HD')

#Rename the sample names to 


```


```{r}



#This code block is to load the required files to plot NSTI vs relative abundance
order_list = c('A5','B1','B2','B7')
abs_count_otu_qza <-  '../dada2/filtered_table.qza'

taxa_vsearch_qza <- '../taxonomy_vsearch/classification.qza'

fam_cols <- c("#341b74","#c7ac92", "#92dce5", "#a06b9a", "#94da1b", "#662e00", "#4a205a", "#008dd5", "#ecb309", "#ff5c6c", "#147b1e", "#353A46", "#f45d01", "#c11f96", "#848fa5", "#bf2231", "#fdfd9b", "#2bb6a8", "#af6e77", "#e08115")

#Try generating Phyloseq object
phyloseq_qza <- qza_to_phyloseq(features = abs_count_otu_qza, taxonomy = taxa_vsearch_qza, metadata = '../read-files/metadata.tsv')


sample_names(phyloseq_qza) <- factor(sample_data(phyloseq_qza)$Sample.name, levels=order_list)

#Phylum
phy_phyl_abs <- phyloseq_qza %>% tax_glom(.,"Phylum") %>% aggregate_taxa(level = "Phylum")  
abund_plot_phyl <- plot_composition(x = phy_phyl_abs,otu.sort = 'abundance') + 
  scale_fill_manual(values = fam_cols) +  ylab('Absolute read counts')
abund_plot_phyl$data$Sample <- factor(abund_plot_phyl$data$Sample, levels=order_list)
  
  phy_phyl_rel <- phyloseq_qza %>% tax_glom(.,"Phylum") %>% aggregate_taxa(level = "Phylum")  %>% microbiome::transform(transform = 'compositional')
  rel_plot_phyl <- plot_composition(x = phy_phyl_rel,otu.sort = 'abundance') + 
    scale_fill_manual(values = fam_cols) +  ylab('Relative Abundance')
  rel_plot_phyl$data$Sample <- factor(rel_plot_phyl$data$Sample, levels=order_list)


#Family
phy_fam_abs <- phyloseq_qza %>% tax_glom(.,"Family") %>% aggregate_taxa(level = "Family")  
abund_plot_fam <- plot_composition(x = phy_fam_abs,otu.sort = 'abundance') + 
  scale_fill_manual(values = fam_cols) +  ylab('Absolute read counts')
abund_plot_fam$data$Sample <- factor(abund_plot_fam$data$Sample, levels=order_list)
  

  phy_fam_rel <- phyloseq_qza %>% tax_glom(.,"Family") %>% aggregate_taxa(level = "Family")  %>% microbiome::transform(transform = 'compositional')
  rel_plot_fam <- plot_composition(x = phy_fam_rel,otu.sort = 'abundance') + 
    scale_fill_manual(values = fam_cols) +  ylab('Relative abundance')
  rel_plot_fam$data$Sample <- factor(rel_plot_fam$data$Sample, levels=order_list)


#Genus
phy_gen_abs <- phyloseq_qza %>% tax_glom(.,"Genus") %>% aggregate_taxa(level = "Genus")  
abund_plot_gen <- plot_composition(x = phy_gen_abs,otu.sort = 'abundance') + 
  scale_fill_manual(values = fam_cols) +  ylab('Absolute read counts')

abund_plot_gen$data$Sample <- factor(abund_plot_gen$data$Sample, levels=order_list)
    
    
    phy_gen_rel <- phyloseq_qza %>% tax_glom(.,"Genus") %>% aggregate_taxa(level = "Genus")  %>% microbiome::transform(transform = 'compositional')
    rel_plot_gen <- plot_composition(x = phy_gen_rel,otu.sort = 'abundance') + 
      scale_fill_manual(values = fam_cols) + ylab('Relative abundance')
    rel_plot_gen$data$Sample <- factor(rel_plot_gen$data$Sample, levels=order_list)


# 
# stacked_fam_gen_plots <- ggarrange(otu_plots_fam, otu_plots_gen, nrow=2, aligh = 'v')

# plot(stacked_fam_gen_plots)

```

```{r}
library(svglite)
otu_plots_gen <- ggarrange(abund_plot_gen, rel_plot_gen, ncol = 2, align ='h', common.legend = T, legend="right")

pdf(file = "./out/16S-plots-stacked-Genus.pdf",   
    width = 7, 
    height = 6) 
  
plot(otu_plots_gen)
otu_plots_phyl <- ggarrange(abund_plot_phyl, rel_plot_phyl, ncol = 2, align ='h', common.legend = T, legend="right")



pdf(file = "./out/16S-plots-stacked-Family.pdf",   
    width = 7, 
    height = 6) 

final_summ_plot <- ggarrange(otu_plots_phyl, otu_plots_gen, ncol = 1, nrow = 2, align = "hv")

plot(final_summ_plot)

ggsave('otu_plot_phyl.svg', otu_plots_phyl,  device = "svg", width = 7, height = 15)
ggsave('otu_plot_gen.svg', otu_plots_gen, device = "svg", width = 7, height = 15) 
  ggsave("final_otu_plot.svg", final_summ_plot, device = "svg", width = 14, height = 12)
```
