
```{r}
library(qiime2R)
library(tidyr)
library(microbiome)
library(compositions)
library(reshape2)
library(gplots)
library(RColorBrewer)
library(ggplot2)
library(janitor)
```

```{r}

#This code block is to stage all the files needed for visualization


#Stage taxonomy file using Qiime2R
taxa_vsearch <- parse_taxonomy(taxonomy = read_qza('../taxonomy_vsearch/classification.qza')$data)
taxa_blast <- parse_taxonomy(taxonomy = read_qza('../taxonomy_blast/classification-blast.qza')$data)

#Stage metadata file
sample_metadata <- read.csv('../read-files/metadata.tsv', sep='\t')

#Stage unstratified/strat kegg pathway files
unstrat_kegg <- read.csv('../picrust2-strat/out/KEGG_pathways_out/path_abun_unstrat.tsv', sep = '\t')
strat_kegg <- read.csv('../picrust2-strat/out/KEGG_pathways_out/path_abun_strat.tsv', sep = '\t')

#Stage unstrat/strat KEGG ortholog files 
ko_unstrat <- read.csv('../picrust2-strat/out/KO_metagenome_out/unstrat_ko_metagenome.tsv', sep = '\t')
ko_strat <- read.csv('../picrust2-strat/out/KO_metagenome_out/strat_ko_metagenome.tsv', sep = '\t')

#Open selected KOs (for plotting relevant Kegg pathways)
deg_kos <- read.csv('./select_kos.txt', sep = '\t', header = TRUE)
rownames(deg_kos) <- deg_kos[,1]
deg_kos[,1] <- NULL

#open the pathway to ko file
pth_2_ko <- read.csv('~/bioinfo_pipelines/picrust2-2.4.2/picrust2/default_files/pathway_mapfiles/KEGG_pathways_to_KO.tsv', header =FALSE, sep='\t') 
#Convert empty strings as NA
pth_2_ko[pth_2_ko==""] <- NA 

#Filter to only Select KOs

pth_2_ko <- pth_2_ko[pth_2_ko$V1 %in% rownames(deg_kos),]
rownames(pth_2_ko) <- pth_2_ko[,1]
pth_2_ko[,1] <- NULL

#Retrieve Sample ids so I can call it as a variable 
sample_ids <- c(as.character(sample_metadata$Sample.id))


```


```{r}

#This code block is to load the required files to plot NSTI vs relative abundance
abs_count_otu <-  read_qza(file = '../dada2/table.qza')$data
rel_count_otu <- as.data.frame(apply(abs_count_otu,2,function(x){x/sum(x)}))
nsti_asvs <- as.data.frame(read.csv('../picrust2-strat/out/marker_predicted_and_nsti.tsv.gz', sep='\t', row.names = 1))


rel_nsti <- merge(nsti_asvs,rel_count_otu,by="row.names",all.x=TRUE)

samps <- c('HC-DS','HC-XY','HC-HX','HC-HD')
rel_nsti <- melt(rel_nsti, measure.vars = samps, variable.name = 'sample.id', value.name = 'rel.abundance')


#Group accding to consortia



ggplot_rel_plot <- ggplot(data=rel_nsti,mapping= aes(x=metadata_NSTI, y=rel.abundance, color=sample.id)) + geom_point()

plot(ggplot_rel_plot)

#No need to present this. This is just a diagnostic plot
```

```{r}
#Generate Heatplot from unstratified KEGG plots


row.names(unstrat_kegg) <- unstrat_kegg$pathway
unstrat_kegg_no_desc <- unstrat_kegg[,2:ncol(unstrat_kegg)]


unstr_keg_mat <- data.matrix(unstrat_kegg_no_desc,rownames.force = TRUE)
#get relative abundance using adorn_percentages  (From Janitor package)
rel_abun_unstrkeg <- adorn_percentages(unstrat_kegg_no_desc, denominator = "col")



```


```{r}


gen_mat <- function(df, ids){
  df_mat <- as.matrix(select(df, ids))
  return(df_mat)
}

#Generate Heatplot from unstratified KEGG plots

#Filter unstrat based on deg_kos table

unstr_keg_filtered <- subset(unstrat_kegg_no_desc, unstrat_kegg$pathway %in% row.names(deg_kos)) 


#Plot heatmap based only on degradation
pdf(file = "./out/KEGG-pathway-heatmap.pdf",   
    width = 8, 
    height = 11) 

heatmap.2(gen_mat(unstr_keg_filtered,sample_ids), 
          labRow = unstr_keg_filtered$description, 
          key=TRUE, keysize=1, symkey=T, lhei = c(1,7), 
          density.info = "none", dendrogram ="none", trace='none', scale='row',
          margins=c(10,24))

dev.off()
```

```{r}
#Generate Heatplot from unstratified KOs per pathway in deg_kos
#I'll generate one first then convert that to a function


#This works na, what I did was generate a list of dataframes first then concatenated them afterwards. I can reuse the degdfs to generate individual 

degdf_list <- list()

for(i in 1:nrow(pth_2_ko)){
  fun_list <- na.omit(unlist(pth_2_ko[i,]))
  fun_df <- subset.data.frame(ko_unstrat, function. %in% fun_list)
  fun_df$pathway <- rownames(pth_2_ko[i,])
  degdf_list[[i]] <-fun_df
}

#Concatenate all dfs in degdf list
deg_df = do.call(rbind, degdf_list)
#The pathway column needs to be categorical pala para mag-work lol
deg_df$pathway <- as.factor(deg_df$pathway)
#reset index
rownames(deg_df) <- NULL


```

```{r}

#Define my color palettes
#I just copied this from some guy's Stackx answer haha
c16 <- c("dodgerblue2", "#E31A1C", "green4", "#6A3D9A", "#FF7F00", "black", "gold1", "skyblue2", "palegreen2", "#FDBF6F", "gray70", "maroon", "orchid1", "darkturquoise", "darkorange4", "brown") 

heat_palette <- colorRampPalette(brewer.pal(3,'YlOrBr'))(n = 100)



#Set Pathways as factor

pdf(file = "./out/Per-gene-heatmap.pdf",   
    width = 8, 
    height = 11) 

heatmap.2((gen_mat(deg_df, sample_ids)), 
          labRow = deg_df$description, labCol = sample_metadata$treatment, 
           key=TRUE, keysize=1, symkey=T, lhei = c(1,7), 
          Rowv = FALSE, Colv = FALSE,
          density.info = "none", dendrogram ="none", trace='none', scale='row',
          margins=c(10,30),
          col = heat_palette,
          RowSideColors=c16[as.numeric(deg_df$pathway)], na.color = 'black' )

legend("left",      
    legend = unique(deg_kos[deg_df$pathway,'description']),
    col = c16[unique(deg_df$pathway)], 
    inset = c(-0.11,0),
    lty= 1,             
    lwd = 5,
    pch =14,
    cex=.3,
    xpd=T
    )

dev.off()

```

```{r}

#Generate Cumulative relative abundances per consortia for specific degradation pathways


#Filter then convert to long form so it's easier to plot
filt_rel_abun <- subset(rel_abun_unstrkeg, rownames(rel_abun_unstrkeg) %in% rownames(deg_kos)) %>% melt(measure.vars = sample_ids, variable.name = 'sample.id', value.name ='rel.abundance') %>% mutate(across(description, factor))

colors <- brewer.pal(11, "Paired")


pdf(file='./out/Deg-pathway-relabunds.pdf', width = 5, height = 5)


ggplot(filt_rel_abun, aes(x=sample.id, y=rel.abundance, fill=description)) +   geom_bar(position="stack", stat="identity") +
scale_fill_manual(values=colors)

dev.off()

```